src/project-manager.ts

```diff
@@ -858,16 +858,36 @@ export class ProjectConfiguration {
 	 * the whole project or in some files
 	 */
 	syncProgram(childOf = new Span()): void {
-		const span = childOf.tracer().startSpan('Sync program', { childOf });
+		const syncProgramSpan = childOf.tracer().startSpan('Sync program', { childOf });
```

Comment:
there should always be only one span per function for readability. Operations that are worth to be traced should be extracted into a new function